sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableProvider.java

```diff
@@ -93,4 +93,18 @@ default Transform[] inferPartitioning(CaseInsensitiveStringMap options) {
   default boolean supportsExternalMetadata() {
     return false;
   }
+
+  /**
+   * Returns true if the source supports defining generated columns upon table creation in SQL.
+   * When false: any create/replace table statements with a generated column defined in the table
+   * schema will throw an exception during analysis.
+   *
+   * A generated column is defined with syntax: {@code colName colType GENERATED ALWAYS AS (expr)}
+   * The generation expression is stored in the column metadata with key "generationExpression".
+   *
+   * Override this method to allow defining generated columns in create/replace table statements.
+   */
+  default boolean supportsGeneratedColumnsOnCreation() {
```

Comment:
We can just call this method `supportsGeneratedColumns`, so that we don't need to add a new API if we want to extend generated columns in other statements in future.

Reply:
Couldn't this be an issue though if we add the syntax to another statement (for example alter table add column) and the catalog/data source (whatever we land on) hasn't updated their implementation yet to accomodate for it?